### PR TITLE
feat(backend): cron saver

### PR DESCRIPTION
**Pull request by Ximaz**

### CODE REVIEW

**Comment by Clement-Z4RM** [comment]
For merges between backend branches, maybe only request review from people working on the backend (so Elouan).
If I review, I'll do it quickly and it's so useless.

**Comment by Ximaz** [comment]
> For merges between backend branches, maybe only request review from people working on the backend (so Elouan). If I review, I'll do it quickly and it's so useless. All the more idk backend code / Nest so I can't really "review".

It's more about being aware of the changes. I'm not talking about telling whether the code is good or bad, but rather see new file types, new / updated endpoints, and so on. That's why I included part of the frontend. Because you will have to deal with those too.

File: common/src/types/oauth/oauth.interface.d.ts [hunk 1]
```diff
@@ -11,9 +11,11 @@ export declare abstract class OAuth {
     abstract readonly clientSecret: string;
     abstract readonly redirectUri: string;
     abstract getOAuthUrl(state: string): string;
-    abstract getCredentials(code: string): Promise<OAuthCredentials>;
-    abstract saveCredentials(userId: Pick<User, "id">["id"], credentials: OAuthCredentials): Promise<number>;
-    abstract loadCredentials(userId: Pick<User, "id">["id"]): Promise<OAuthCredentials[]>;
-    abstract refreshCredentials(oauthCredentials: OAuthCredentials): Promise<OAuthCredentials>;
-    abstract revokeCredentials(oauthCredentials: OAuthCredentials): Promise<void>;
+    abstract getCredentials(code: string): Promise<OAuthCredential>;
```
Comment (Clement-Z4RM) [inline common/src/types/oauth/oauth.interface.d.ts:14]: Couldn't it cause problems with requied get... field when we'll use this type on frontend?

File: common/src/types/area/interfaces/area.interface.d.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    service: string;
+    method: string;
+    config: ActionDescription;
+}
+export interface AreaReaction {
+    service: string;
+    method: string;
+    config: ReactionDescription;
+}
+export interface AreaTask {
```
Comment (Clement-Z4RM) [inline common/src/types/area/interfaces/area.interface.d.ts:14]: What's the difference between AreaTask and Area?

**Review by Clement-Z4RM — APPROVED** [review]

File: common/src/types/oauth/oauth.interface.d.ts (hunk 1 above)
Comment (Ximaz) [reply common/src/types/oauth/oauth.interface.d.ts:14]: I'm not sure to understand the question.

File: common/src/types/area/interfaces/area.interface.d.ts (hunk 2 above)
Comment (Ximaz) [reply common/src/types/area/interfaces/area.interface.d.ts:14]: Basically, the Area object is the DB representation of an AreaTask, excepted that an AreaTask object is loaded with both the functions to call the Action and Reaction, but also the functions to deal with OAuth credentials (like refreshing the credentials for instance.)

File: common/src/types/oauth/oauth.interface.d.ts (hunk 1 above)
Comment (Clement-Z4RM) [reply common/src/types/oauth/oauth.interface.d.ts:14]: If we create an OAuthCredential, could we not have TS warnings if we don't set function properties like getCredentials?

**Review by ElouanR — APPROVED** [review]

File: common/src/types/oauth/oauth.interface.d.ts (hunk 1 above)
Comment (Ximaz) [reply common/src/types/oauth/oauth.interface.d.ts:14]: You will not create an `OAuthCredential`. That type is exclusively for backend services. You won't have to deal with it.

Comment (Clement-Z4RM) [reply common/src/types/oauth/oauth.interface.d.ts:14]: So why it's in common?

**Pull request merged**